index.js

```diff
@@ -166,22 +165,23 @@ class HandlebarsRenderer {
      * @param {Object} A set of templates to register with handlebars
      */
     addTemplates(templates) {
-        _.each(templates, (template, path) => {
-            // Don't do this work twice, first one wins.
+        const paths = Object.keys(templates);
+
+        for (let i = 0; i < paths.length; i++) {
+            const path = paths[i];
+
             if (typeof this.handlebars.partials[path] !== 'undefined') {
                 return;
             }
-
             // Check if it is a precompiled template
             try {
-                template = this._tryRestoringPrecompiled(template);
+                const template = this._tryRestoringPrecompiled(templates[path]);
```

Comment:
You should move the comment down if you want to handle it this way.

```
try {
  // Check if it is a precompiled template
  const template = this._tryRestoringPrecompiled(templates[path]);

  // Register it with handlebars
  this.handlebars.registerPartial(path, template);
} catch(e) {
}
```

Reply:
Or if you prefer

```
try {
  // Register it with handlebars after checking that it is a precompiled template
  const template = this._tryRestoringPrecompiled(templates[path]);
  this.handlebars.registerPartial(path, template);
} catch(e) {
}
```

Reply:
I went the first way.  I thought it kind of looked weird before and this makes sense.